### ISSUE TITLE
Updating with install script trashes html client default welcome message when an ampersand is present

### ISSUE THREAD

**Issue by AlphaDE** [bigbluebutton/bbb-install#399]
My welcome message shown in the chat window when joining a conference is

`defaultWelcomeMessage=Willkommen im Raum <b>%%CONFNAME%%</b>!<br><br>Um Ihre Audioverbindung zu starten, klicken Sie auf den Telefonh&ouml;rerbutton.<br><br>Welcome to <b>%%CONFNAME%%</b>!<br><br>To join the audio bridge click the phone button.`

The html representation of the German umlaut oe (&ouml;) irritates the update script. When running the update script, the default messages is inserted into the defaultWelcomeMessage string at the location of the & (ampersand) symbol and then looks like

`defaultWelcomeMessage=Willkommen im Raum <b>%%CONFNAME%%</b>!<br><br>Um Ihre Audioverbindung zu starten, klicken Sie auf den TelefonhdefaultWelcomeMessage=Welcome to <b>%%CONFNAME%%</b>!<br><br>For help on using BigBlueButton see these (short) <a href="https://www.bigbluebutton.org/html5"><u>tutorial videos</u></a>.<br><br>To join the audio bridge click the phone button.  Use a headset to avoid causing background noise for others.ouml;rerbutton.<br><br>Welcome to <b>%%CONFNAME%%</b>!<br><br>To join the audio bridge click the phone button.`

It would be great if this could be fixed.

**Comment by ffdixon** [comment]
Recommend you use the ability in 2.3 to [override the configuration file](https://docs.bigbluebutton.org/admin/configuration-files#local-overrides-for-configuration-settings) for `bigbluebutton.properties`.

For example if you add to `/etc/bigbluebutton/bbb-web.properties` the following line

```
defaultWelcomeMessage=Willkommen im Raum <b>%%CONFNAME%%</b>!<br><br>Um Ihre Audioverbindung zu starten, klicken Sie auf den Telefonh&ouml;rerbutton.<br><br>Welcome to <b>%%CONFNAME%%</b>!<br><br>To join the audio bridge click the phone button.
```

Then your welcome message will persist between upgrades.

**Comment by AlphaDE** [comment]
@ffdixon Thanks for pointing me to that option.

I though recommend to check the search&replace there.

https://docs.bigbluebutton.org/admin/customize.html should be updated with the option as that page recommends to update /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties

**Comment by ffdixon** [comment]
Agreed -- we'll update the docs as it's now much better to add these parameters to the override file.  We'll leave this issue open until the docs are updated.

To help out the project, feel free to send a pull request to make the necessary updates.  See https://github.com/bigbluebutton/bigbluebutton.github.io.

**Comment by michi-80337** [comment]
I'm absolutely not sure yet, because I still haven't found the point where it happens in all the scripts involved.
But I suppose, the error happens when the occurrences of the value from bigbluebutton.web.serverURL are modified in the bigbluebutton.properties file. The ampersand has a special meaning in sed, and it will be replaced by the search string. Beside the ampersand one has to keep an eye on the backslash and the delimiter char used (in most cases a slash). In example, following code would print the second line as "--- Hello World ---":

    sed -e "s/Hello world/--- & ---/" <<HERE
    Blah blubb
    Hello world
    Tralala
    HERE

@AlphaDE: What happens, if you remove the `<a ...>` and the `</a>` tags from the message, so that there is no URL in the text anymore? Will the message still be doubled by the update?
